include/alpaka/mem/view/AccessorAdaptor.hpp

```diff
@@ -0,0 +1,169 @@
+/* Copyright 2022 Andrea Bocci
+
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED “AS IS” AND ISC DISCLAIMS ALL WARRANTIES WITH
+ * REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL ISC BE LIABLE FOR ANY
+ * SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR
+ * IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#pragma once
+
+#include <alpaka/dim/Traits.hpp>
+#include <alpaka/extent/Traits.hpp>
+#include <alpaka/mem/view/ViewAccessor.hpp>
+
+#include <sstream>
+#include <type_traits>
+
+namespace alpaka
+{
+    namespace internal
+    {
+        template<typename TView>
+        struct AccessorAdaptor
```

Comment:
I think the type's name is misleading. First, it has very little to do with `alpaka::experimental::Accessor`, which is also already a bad name. I would try to avoid the name "accessor" because it has a very broad meaning. Also the term "accessor" in C++23 will have a different meaning (see `std::mdspan`). Second, the type does not strictly implement an adaptor, but a mixin.

Since this CRTP mixin is intended for views, you could for example name it `ViewAccessOps`.

Reply:
> it has very little to do with `alpaka::experimental::Accessor`

well, it implements the same interface, basically allowing to use Buffer or View directly in place of an Accessor

> ... which is also already a bad name

this is probably outside the scope of his PR.





> Since this CRTP mixin is intended for views, you could for example name it `ViewAccessOps`.

Other suggestions for a better name ?